cli/listener/src/main.rs

```diff
@@ -452,6 +453,32 @@ fn get_date_params<'a, I: IntoIterator<Item = (Cow<'a, str>, Cow<'a, str>)>>(
                 })?;
                 end_opt = Some(t);
             }
+            "days" => {
+                let days = v.parse::<i64>().map_err(|_| {
+                    ResponseError::BadRequest(
+                        "Invalid number of days in 'days' query parameter. \
+                    Expected e.g. '30'.",
+                    )
+                })?;
+                let end = chrono::Utc::now();
+                let begin = end
+                    .checked_sub_signed(chrono::Duration::days(days))
+                    .ok_or_else(|| {
+                        ResponseError::BadRequest(
+                            "Invalid number of days in 'days' query parameter. \
+                    Expected e.g. '30'.",
```

Comment:
If we get here, `days` is already a valid integer; the reason for the `checked_sub_signed` to fail is when it overflows. The reason for that is when you pass a ridiculously large number. We can improve the message by saying something like “Date range too large”.